test/util/page-number-functions.spec.js

```diff
@@ -0,0 +1,77 @@
+const { expect } = require('chai')
+const { findPageNumbers, findFirstPage, removePageNumber } = require('../../lib/util/page-number-functions')
+
+describe('functions: findPageNumbers', () => {
+  it('Search, coerce and store page number', () => {
+    const array = [{ str: '1' }, { str: 'test' }, { str: 'how' }, { str: 'to' }, { str: 'find' }, { str: 'page' }, { str: '3' }, { str: 'number' }]
+    expect(findPageNumbers({}, 3, array)).to.eql({ 3: [1, 3] })
+  })
+})
+
+describe('functions: findFirstPage', () => {
+  it('Return first page index and number with completed Hashtable', () => {
+    const hashTable = {
```

Comment:
This is really a dictionary (okay, a JavaScript object to be more precise); JavaScript objects aren't quite hash tables

Reply:
Interesting I didn't know that. I thought they were the same. I have changed the variable name to `object`.

Reply:
keyphrase "aren't quite"..